src/index.c

```diff
@@ -734,8 +734,11 @@ void git_index_entry__init_from_stat(
 {
 	entry->ctime.seconds = (git_time_t)st->st_ctime;
 	entry->mtime.seconds = (git_time_t)st->st_mtime;
-	/* entry->mtime.nanoseconds = st->st_mtimensec; */
-	/* entry->ctime.nanoseconds = st->st_ctimensec; */
+#if !defined(GIT_WIN32) && !defined(__APPLE__)
+	/* Apple and Windows doesn't provide these struct stat fields. */
```

Comment:
Furthermore, shouldn't we be reading these only if `GIT_USE_NSEC` is defined? I think it would turn out cleaner if we only offer the `USE_NSEC` option when building under Linux and remove the OS checks inside the libgit2 code. If we add support for this in Windows, we'd add it inside the compat layer as well, so we likely wouldn't have to change the code here.